GOVERNANCE.md

```diff
@@ -39,18 +39,30 @@ below.
 - Resolution of bugs triaged to a package/feature
 - Regularly review pull requests to the pkg subsystem
 
-Contributors who are interested in becoming a maintainer, if performing these
+### Nomination and retiring of maintainers
+
+[Maintainers](./MAINTAINERS) file on the `main` branch reflects the latest
+state of project maintainers. Changes to list of maintainers should be done by
+opening a pull request and CCing all the existing maintainers.
+
+Contributors who are interested in becoming a maintainer, if performing relevant
 responsibilities, should discuss their interest with the existing maintainers. New
 maintainers must be nominated by an existing maintainer and must be elected by a
 supermajority of maintainers with a fallback on lazy consensus after three business weeks
-inactive voting period and as long as two maintainers are on board. Maintainers can be
-removed by a supermajority of the maintainers and moved to emeritus status.
-
-Life priorities, interests, and passions can change. If a maintainer needs to step
-down, inform other maintainers about this intention, and if possible, help find someone
-to pick up the related work. At the very least, ensure the related work can be continued.
-Afterward, create a pull request to remove yourself from the [MAINTAINERS](./MAINTAINERS)
-file.
+inactive voting period and as long as two maintainers are on board. 
+
+Life priorities, interests, and passions can change. Maintainers can retire and
+be removed from [maintainers](./MAINTAINERS) list and added to
+[emeritus](./README.md#etcd-emeritus-maintainers). If a maintainer needs to step
+down, they should inform other maintainers, if possible, help find someone to
+pick up the related work. At the very least, ensure the related work can be
+continued. Afterward they can remove themselves from list of existing maintainers.
+
+If a maintainer is not been performing their duties for period of 12 months,
```

Comment:
Let's also add why such a step is needed -  "If a maintainer has not been performing their duties for period of 12 months, they can be moved to  emeritus status by active maintainers to gain better understanding of recruiting need for new maintainers."

Reply:
Also, I think we should add something like this after the above sentence - "Before such a change, an inactive maintainer should be notified through email or slack by active maintainers considering GitHub notifications sometimes go unnoticed."

Reply:
Ad 1, added motivation to maintain the list to first section.

Ad 2, I think email is enough as all maintainers are required to publish their preferred email contact in ./MAINTAINERS.

Reply:
Thanks @serathius and yes, I agree with both the points - email is good enough, and we should remove IRC reference.